compiler/noirc_frontend/src/debug/mod.rs

```diff
@@ -0,0 +1,610 @@
+use crate::parser::{parse_program, ParsedModule};
+use crate::{
+    ast,
+    ast::{Path, PathKind},
+    parser::{Item, ItemKind},
+};
+use noirc_errors::{Span, Spanned};
+use std::collections::HashMap;
+use std::collections::VecDeque;
+
+#[derive(Debug, Clone)]
+pub struct DebugState {
+    pub variables: HashMap<u32, String>, // var_id => var_name
+    pub field_names: HashMap<u32, String>,
+    next_var_id: u32,
+    next_field_name_id: u32,
+    scope: Vec<HashMap<String, u32>>, // var_name => var_id
+    pub enabled: bool,
+}
+
+impl Default for DebugState {
+    fn default() -> Self {
+        Self {
+            variables: HashMap::default(),
+            field_names: HashMap::default(),
+            scope: vec![],
+            next_var_id: 0,
+            next_field_name_id: 1,
+            enabled: true, // TODO
```

Comment:
Unresolved TODO.

Reply:
Removed the field altogether since it was not being used.